fem/bilininteg.cpp

```diff
@@ -2257,26 +2304,51 @@ void VectorDiffusionIntegrator::AssembleElementVector(
    for (int i = 0; i < ir->GetNPoints(); i++)
    {
       const IntegrationPoint &ip = ir->IntPoint(i);
+      el.CalcDShape(ip, dshape);
 
       Tr.SetIntPoint(&ip);
-      CalcAdjugate(Tr.Jacobian(), Jinv);
-      w = ip.weight / Tr.Weight();
-      if (Q)
+      double w = Tr.Weight();
+      w = ip.weight / (square ? w : w*w*w);
+
+      if (VQ)
       {
-         w *= Q->Eval(Tr, ip);
+         VQ->Eval(vcoeff, Tr, ip);
+         for (int k = 0; k < vdim; ++k)
+         {
+            Mult_a_AAt(w*vcoeff(k), dshapedxt, pelmat);
```

Comment:
I guess you don't need to construct `pelmat`, only to provide it's action 
```
vec_out += w*coeff(k)*dshapeext * dshapeext^T vec_in
```

Reply:
I agree but I don't see a better way to do this. A custom `Mult_a_AAt_V` could be written but I've rewritten to at least reduce the number of operations.

Reply:
I have accepted the PR since it does not seem you want to further optimize this and you probably do not need it.

Reply:
I'm interested in further optimizing but I really don't see how without introducing a new operator somehow.

In your pseudocode, the action is
`v_o += alpha * D*D^T * v_i`
which is the product of a scalar, a matrix and its transpose, and a vector. Hence, `Mult_a_AAt_V`.

Functions like
```cpp
mfem::Vector mfem::Vector::operator*(double a);
mfem::DenseMatrix mfem::DenseMatrix::operator*(double a);
void mfem::DenseMatrix::Mult_MultTranspose(const mfem::Vector& v_i, mfem::Vector& v_o);
```
would give more options but I don't see how to implement this accumulation with the operators at hand.

Otherwise, I could use `pelmat` with `P=D*D^T` like I'm doing now and then `P.AddMult_a(w*vcoeff(k), vec_out, vec_in)`.